content/Wrath.SoundVoiceModTemplate/{SourceName}/Main.cs

```diff
@@ -0,0 +1,935 @@
+        return true;
+    }
+
+    public static void OnGUI(UnityModManager.ModEntry modEntry) {
+
+    }
+
+    [HarmonyPatch]
+    public static class Soundbanks {
+        public static readonly HashSet<uint> LoadedBankIds = [];
+
+        [HarmonyPatch(typeof(AkAudioService), nameof(AkAudioService.Initialize))]
+        [HarmonyPostfix]
+        public static void LoadSoundbanks() {
+            var banksPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+            try {
+                log.Log($"Add soundbank base path {banksPath}");
+                AkSoundEngine.AddBasePath(banksPath);
+
+                foreach (var f in Directory.EnumerateFiles(banksPath, "*.bnk")) {
+                    var bankName = Path.GetFileName(f);
+                    var akResult = AkSoundEngine.LoadBank(bankName, out var bankId);
+
+                    if (bankName == "Init.bnk")
+                        throw new InvalidOperationException("Do not include Init.bnk");
+
+                    if (akResult == AKRESULT.AK_BankAlreadyLoaded)
+                        continue;
+
+                    log.Log($"Loading soundbank {f}");
+
+                    if (akResult == AKRESULT.AK_Success) {
+                        LoadedBankIds.Add(bankId);
+                    } else {
+                        log.Error($"Loading soundbank {f} failed with result {akResult}");
+                    }
+                }
+            } catch (Exception e) {
+                log.LogException(e);
+                UnloadSoundbanks();
+            }
+        }
+
+        public static void UnloadSoundbanks() {
+            foreach (var bankId in LoadedBankIds) {
+                try {
+                    AkSoundEngine.UnloadBank(bankId, IntPtr.Zero);
+                    LoadedBankIds.Remove(bankId);
+                } catch (Exception e) {
+                    log.LogException(e);
+                }
+            }
+        }
+
+        [HarmonyPatch(typeof(BlueprintsCache), nameof(BlueprintsCache.Init))]
+        [HarmonyPostfix]
+        static void AddAsksListBlueprint()
+        {
+            LocalizationManager.CurrentPack.PutString("{SourceName}", "{Description}");
+
+            var blueprint = new BlueprintUnitAsksList
+            {
+				// Every mod requires its own unique GUID. Autogenerated on template creation.
+				AssetGuid = new(System.Guid.Parse("{ModGUID}")),
+                name = "{SourceName}_Barks",
+                DisplayName = new() { m_Key = "{SourceName}" }
+            };
+
+            blueprint.ComponentsArray =
+            [
+                new UnitAsksComponent()
+            {
+                OwnerBlueprint = blueprint,
+
+                // Since the blueprint is added manually by the mod, remove the usual reference
+                // to the bank name to prevent a Wwise "already loaded" error.
+                SoundBanks = [],
+                PreviewSound = "{SourceName}_Test",
+                Aggro = new()
```

Comment:
That's a lot of duplicate code here; do you think you could move instantiating into a function (with stuff that changes as parameter)?